### PR TITLE
Periodic boundary conditions: Fix orientation problem

### LINKED ISSUE #16170: Discrepancy in Periodicity Handling with Serial and Distributed Triangulations
For certain meshes (having periodic boundaries) generated with external meshing programs such ash Gmsh, the `dealii::parallel::distributed::Triangulations::add_periodicity()` seems to throw an assertion, while there is not issue in detecting the periodic faces with the serial Triangulation.

A minimal working example, with detailed output logs, error messages and some example mesh files can be found [on this mailing list thread](https://groups.google.com/d/msgid/dealii/6fa10380-8535-4f29-b5cf-7cd40fd62e21n%40googlegroups.com?utm_medium=email&utm_source=footer).

### PR DESCRIPTION
**Pull request by kronbichler**
This should fix the issue reported in #16170: It turned out that we aimed to compensate for the orientation along the periodic boundaries, but in 2D we would not actually reorder the vertices because `GeometryInfo<dim>` would assume that the triangulation is oriented consistently.

Closes #16170

### CODE REVIEW

File: source/distributed/tria.cc [hunk 1]
```diff
@@ -2623,41 +2623,33 @@ namespace parallel
...
                     // cell[0] into account
-                    const unsigned int vface0 =
-                      GeometryInfo<dim>::standard_to_real_face_vertex(
-                        v, orientation, flip, rotation);
-                    const unsigned int vi0 =
-                      topological_vertex_numbering[cell_1->face(face_no_1)
-                                                     ->vertex_index(vface0)];
+                    const unsigned int vface1 =
+                      cell_1->reference_cell().standard_to_real_face_vertex(
+                        v, face_no_1, combined_orientation);
```
Comment (kronbichler) [inline source/distributed/tria.cc:2643]: This is the relevant change, all other changes are just cleanups in the variable names and switch to range-based for loops.

**Review by masterleinad — APPROVED** [review]
Can you also add a test?

File: source/distributed/tria.cc [hunk 2]
```diff
@@ -2623,41 +2623,33 @@ namespace parallel
         // {0,0,0,0,0,0,0,0}.
         using cell_iterator =
           typename Triangulation<dim, spacedim>::cell_iterator;
-        typename std::map<std::pair<cell_iterator, unsigned int>,
-                          std::pair<std::pair<cell_iterator, unsigned int>,
-                                    unsigned char>>::const_iterator it;
-        for (it = tria.get_periodic_face_map().begin();
-             it != tria.get_periodic_face_map().end();
-             ++it)
+        for (const auto it : tria.get_periodic_face_map())
```
Comment (tamiko) [inline source/distributed/tria.cc:2626]:
```suggestion
        for (const auto &it : tria.get_periodic_face_map())
```

**Comment by tamiko** [comment]
@kronbichler will add a test and change the for loop later today.

**Comment by kronbichler** [comment]
I now added a test case, adapted from the case provided by @KParas in #16170. @KParas could you please check if this fixes the issue for you? Sorry for taking some time, we just got to it now with the release preparations.

**Review by tamiko — APPROVED** [review]

**Pull request merged**